### PR TITLE
fix(prover): implement eth_getBlockNumber verified request handler

### LINKED ISSUE #5796: Verification of `eth_getBlockNumber` for prover should not forward request
### Describe the bug

Currently we documented that `eth_blockNumber` will be forwarded to EL. This may cause to return the wrong block number which is not synced with the light client. 

### Expected behavior

Return the latest block number signed by the current sync-committee. 

### Steps to reproduce

_No response_

### Additional context

_No response_

### Operating system

Linux

### Lodestar version or commit hash

1.9.x

### PR DESCRIPTION
**Pull request by TechFusionData**
Closes ChainSafe/lodestar-prover#7. The prover was forwarding eth_getBlockNumber to the execution layer instead of serving it locally from the PayloadStore, which defeats the purpose of the verified light client proxy.

Changes:
- Add eth_getBlockNumber handler in verified_requests/ that reads the latest block number from proofProvider.getExecutionPayload('latest') and returns it as a hex string
- Register the handler in verifiableMethodHandlers in utils/process.ts
- Add unit tests for happy path and verification-failed error path

Note: Implementation assisted with AI (Claude).

**Motivation**

The prover was forwarding `eth_getBlockNumber` to the execution layer instead of serving it locally, defeating the purpose of the verified light client proxy.

Implements a verified handler for `eth_getBlockNumber` following the same pattern as `eth_getBlockByNumber`. Reads the block number from `proofProvider.getExecutionPayload('latest')` and returns it as a hex string. Includes unit tests for both happy path and verification failure.

Closes #issue_number

**AI Assistance Disclosure**

- [x ] External Contributors: I have read the [contributor guidelines](https://github.com/ChainSafe/lodestar/blob/unstable/CONTRIBUTING.md#ai-assistance-notice) and disclosed my usage of AI below.

Used Claude (Anthropic) to assist with implementation and writing unit tests. The approach, logic, and code were reviewed and understood by me before submission.

### CODE REVIEW

File: packages/prover/src/verified_requests/eth_blockNumber.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  getVerificationFailedMessage,
+} from "../utils/json_rpc.js";
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const eth_blockNumber: ELVerifiedRequestHandler<[], string> = async ({payload, logger, proofProvider}) => {
+  try {
+    const executionPayload = await proofProvider.getExecutionPayload("latest");
+    const blockNumber = numberToHex(executionPayload.blockNumber);
+
+    return getResponseForRequest(payload, blockNumber);
```
Comment (nazarhussain) [inline packages/prover/src/verified_requests/eth_blockNumber.ts]: The verification process is missing in these statements, which is main motivation of the `verified_requests` in `prover`. 

User request for `eth_bl;ockNumber` meant to fetch latest block from the EL nodes. So we should fetch that first and then verify against our CL light-client.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

**Comment by TechFusionData** [comment]
Thanks for the review — you're right, the handler was only reading from the CL without fetching from the EL first. Fixed in the latest commit: now fetches `eth_blockNumber` from the EL, then verifies it against the CL's latest execution payload before returning. Also added `eth_blockNumber` to the `ELApi` type map.

File: packages/prover/src/verified_requests/eth_blockNumber.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    const executionPayload = await proofProvider.getExecutionPayload("latest");
+    const clBlockNumber = executionPayload.blockNumber;
+
+    if (elBlockNumber < clBlockNumber) {
+      throw new Error(
+        `EL block number (${elBlockNumber}) is behind CL block number (${clBlockNumber})`
+      );
+    }
+
+    return getResponseForRequest(payload, numberToHex(elBlockNumber));
```
Comment (nazarhussain) [inline packages/prover/src/verified_requests/eth_blockNumber.ts]: Looking at this whole logic holistically, I feel it's not right. 

- `el < cl` means the EL is stale/inconsistent relative to your trusted CL-derived execution payload
- `el >= cl` means at least the EL is not behind the trusted lower bound

But there’s still a conceptual issue:
- if you return `elBlockNumber` when `el > cl`, that returned value is not actually CL-verified
- it is only “not contradicted by CL”

So looking from the user perspective what he is expecting a **verified latest block number** and that can only mean a CL execution payload block number. 

So we can do it with either 2 ways: 

1. Do not call EL at all and only return what CL verified view is
2. Add drift window e.g. 2 blocks, so if EL is maximum 2 blocks forward then it's ok otherwise we consider that CL is too much behind to even consider it a verified latest block number. 

The second approach seems more reasonable to me.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

**Comment by TechFusionData** [comment]
Good point — you're right that returning EL's block when EL > CL means it's only "not contradicted" by CL, not actually verified. Updated to your second approach: drift window of 2 blocks. The handler now uses the EL response only as a staleness check (rejects if |EL − CL| > 2) and returns the CL-verified block number itself. Tests cover the four edge cases (within window, equal, EL behind CL, CL too stale).

**Comment by wemeetagain** [comment]
prover is no longer hosted here, now in github.com/chainsafe/lodestar-prover
